### PR TITLE
🏃cleanup: enable unparam linter

### LINKED ISSUE #1544: cleanup: enable `unparam` linter
After #1535, we updated the golangci-lint to the latest version and matched the upstream Cluster API format for the linter, it seems there were a bunch of linters missing.

- Head over to [.golanci-lint.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/.golangci.yml) in the root of this repository.
- Remove the linter from the `linters.disable` list.
- Run `make lint-full`.
- Fix the reported errors with the suggestions.
- Submit a PR with the changes.

Feel free to reach out at any time if you need guidance or help.

/kind cleanup
/help
/good-first-issue
/milestone Next
/priority backlog

### PR DESCRIPTION
**Pull request by hypnoglow**
**What this PR does / why we need it**:

This change enables `unparam` linter.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1544

### CODE REVIEW

File: controllers/awsmachine_controller.go [hunk 1]
```diff
@@ -282,7 +282,7 @@ func (r *AWSMachineReconciler) findInstance(scope *scope.MachineScope, ec2svc se
 	return instance, nil
 }
 
-func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope *scope.MachineScope, clusterScope *scope.ClusterScope) (reconcile.Result, error) {
+func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *scope.MachineScope, clusterScope *scope.ClusterScope) (reconcile.Result, error) {
```
Comment (hypnoglow) [inline controllers/awsmachine_controller.go]: Not removing the parameter, assuming that we are looking forward to improve the context propagation.

**Comment by vincepri** [comment]
@hypnoglow Are you able to rebase on master?

**Comment by hypnoglow** [comment]
Rebased

**Pull request merged**